src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/TokenCertificateService.java

```diff
@@ -556,29 +561,82 @@ public void registerAuthCert(String hash, String securityServerAddress) throws C
     }
 
     /**
-     * Send the authentication certificate deletion request to central server
+     * Send the authentication certificate deletion request to central server and set the cert status to
+     * {@link CertificateInfo#STATUS_DELINPROG}
      * @param hash certificate hash
-     * @throws CertificateNotFoundException
+     * @param skipUnregister whether to skip the actual delete request and only change cert status
+     * @throws SignCertificateNotSupportedException
+     * @throws ActionNotPossibleException
      * @throws GlobalConfService.GlobalConfOutdatedException
+     * @throws InvalidCertificateException
+     * @throws KeyNotFoundException
+     * @throws CertificateNotFoundException
+     * @throws NoValidAuthCertificateException
      */
-    public void unregisterAuthCert(String hash) throws CertificateNotFoundException,
-            GlobalConfService.GlobalConfOutdatedException, InvalidCertificateException,
-            SignCertificateNotSupportedException, KeyNotFoundException, ActionNotPossibleException {
+    private void unregisterAuthCertAndMarkForDeletion(String hash, boolean skipUnregister)
+            throws CertificateNotFoundException, GlobalConfService.GlobalConfOutdatedException,
+            InvalidCertificateException, SignCertificateNotSupportedException, KeyNotFoundException,
+            ActionNotPossibleException, NoValidAuthCertificateException {
         CertificateInfo certificateInfo = getCertificateInfo(hash);
         verifyAuthCert(certificateInfo);
         verifyCertAction(PossibleActionEnum.UNREGISTER, certificateInfo, hash);
         SecurityServerId securityServerId = serverConfService.getSecurityServerId();
         try {
-            managementRequestSenderService.sendAuthCertDeletionRequest(securityServerId,
-                    certificateInfo.getCertificateBytes());
+            if (!skipUnregister) {
+                managementRequestSenderService.sendAuthCertDeletionRequest(securityServerId,
+                        certificateInfo.getCertificateBytes());
+            }
             signerProxyFacade.setCertStatus(certificateInfo.getId(), CertificateInfo.STATUS_DELINPROG);
-        } catch (GlobalConfService.GlobalConfOutdatedException | CodedException e) {
+        } catch (CodedException ce) {
+            if (isCausedBySslAuthFailure(ce)) {
+                throw new NoValidAuthCertificateException(ce);
+            }
+            throw ce;
```

Comment:
Frontend could handle this situation so that when `unregisterAuthCertificate` endpoint returns anything else than `http 204` -> a popup will be offered to mark the cert for deletion.

Now I would suggest getting rid of the `CodedException` error handling here (and let our `ExceptionTranslator` handle it as a 500) because we are sending an actual X-Road message and there are a lot of different errors that might occur.

Reply:
Yeah. Maybe the last suggestion, about XroadMessageException, would be good to make our implementation work the same way as old implementation. More about this in slack discussion.

Reply:
There are now separate try-catch-blocks for both `sendAuthCertDeletionRequest` and `setCertStatus` and they provide different exceptions:

`sendAuthCertDeletionRequest` -> `ManagementRequestSendingFailedException` -> `InternalServerErrorException`

`setCertStatus` -> `RuntimeException`